src/Sulu/Component/Rest/ListBuilder/Expression/Doctrine/DoctrineInExpression.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Expression\Doctrine;
+
+use Doctrine\ORM\QueryBuilder;
+use Sulu\Component\Rest\ListBuilder\Doctrine\FieldDescriptor\AbstractDoctrineFieldDescriptor;
+use Sulu\Component\Rest\ListBuilder\Expression\InExpressionInterface;
+
+/**
+ * Represents a IN expression for doctrine - needs a field and an array of values.
+ */
+class DoctrineInExpression extends AbstractDoctrineExpression implements InExpressionInterface
+{
+    /**
+     * Field descriptor used for comparison.
+     *
+     * @var AbstractDoctrineFieldDescriptor
+     */
+    protected $field;
+
+    /**
+     * Array values to compare.
+     *
+     * @var array
+     */
+    protected $values;
+
+    /**
+     * DoctrineInExpression constructor.
+     *
+     * @param AbstractDoctrineFieldDescriptor $field
+     * @param array $values
+     */
+    public function __construct(AbstractDoctrineFieldDescriptor $field, array $values)
+    {
+        $this->values = $values;
+        $this->field = $field;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getStatement(QueryBuilder $queryBuilder)
+    {
+        $paramName = $this->getFieldName() . uniqid(true);
+        $values = $this->filterNullValues($this->getValues());
+        $statement = '';
+
+        if (count($values) > 0) {
+            $queryBuilder->setParameter($paramName, $values);
+            $statement = $this->field->getSelect() . ' IN (:' . $paramName . ')';
+
+            if (array_search(null, $this->getValues())) {
```

Comment:
I can put them in a separate if statement but I still have to know wether the string is empty and if null values are in the array - so not really a difference?